cypress/integration/e2e/application_spec.js

```diff
@@ -0,0 +1,50 @@
+import {DESIGNER} from "../../support/constants"
+describe("Application Spec", () => {
+  beforeEach(() => {
+    cy.login();
+    cy.setReleaseTag();
+    cy.interceptCapabilities();
+    window.localStorage.setItem("tab", 0)
+    cy.setMode(DESIGNER)
+    cy.visit("/extension/meshmap");
+    cy.intercept("/api/provider/extension*").as("extensionFileLoad");
+    cy.wait("@extensionFileLoad", { timeout: 20000 });
+  })
+
+  it("Visit MeshMap Application", () => {
+    cy.get('[data-cy="application-drawer"]').click();
+    cy.contains("Applications")
+    cy.get("#MUIDataTableBodyRow-applications-0").should("be.visible").click(); //convention: MUIDataTableBodyRow + type  + rowIndex
+    cy.wait(2000);
+    cy.get("body").then(body => {
+        if (body.find("[aria-describedby='notistack-snackbar'] #notistack-snackbar").length > 0) {
+          cy.get("[aria-describedby='notistack-snackbar'] #notistack-snackbar").should("not.contain", "Unable to render")
+        }
+      })
+  });
+
+  it("Rename and Saving Application", () => {
+    cy.get("#component-drawer-Application").should('be.visible')
+    cy.get("#design-name-textfield").type("Changed Application Name with cypress");
+    cy.intercept('/api/application').as('applicationSave')
+ 
+    // TODO - Saving Application request intercept need to be fixed.
+
```

Comment:
@Abhishek-kumar09 it is not returning the request, any suggestion to fix it?

Reply:
Application has problems with renaming, you should see the red snackbar with it. That's why the interception fails.

Reply:
Even if the test fails, write the logic correctly, the tests should be correct with their logic.

Reply:
Sure